### ISSUE TITLE
[Bug]: PC0013 false positive on IsolatedStorage.Get inside a table since 1.3.0 (bare-self fallback treats static built-in call as Rec.Get)

### ISSUE THREAD

**Issue by JohnnyUndercover** [ALCops/Analyzers#550]
### Analyzer

PlatformCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

PC0013

### Description

Since ALCops.Analyzers **1.3.0** (released 2026-09-14), PC0013 (`RecordGetProcedureArguments`) reports a false positive for `IsolatedStorage.Get(...)` when the call is placed inside a **table** (procedure or trigger). The same code compiled cleanly with 1.2.0 and nothing else changed (same altool 17.0.34.45391, same sources).

Reported diagnostic:

```
ADConnector.Table.al(190,13): error PC0013: Invalid arguments in call to 'Get("Primary Key": Integer)'
for record 'AHG-BA AD Connector': Too many arguments provided; expected 1 arguments.
```

The flagged statement is `IsolatedStorage.Get(Key, DataScope::Module, Password)`, i.e. the static built-in `IsolatedStorage.Get`, not the record's `Get`.

**Likely root cause:** PR #509 changed `RecordGetProcedureArguments` from

```csharp
if (invocation.Instance?.Type.GetTypeSymbol()?.OriginalDefinition is not ITableTypeSymbol table) return;
```

to

```csharp
if (invocation.Instance.GetReceiverTableType(ctx.ContainingSymbol, out _) is not ITableTypeSymbol table) return;
```

`GetReceiverTableType` treats `Instance == null` as the "bare implicit self" form and falls back to the containing table. However, calls on static built-in types such as `IsolatedStorage` also bind with `Instance == null` (there is no receiver operation for a static type). Inside a table this makes every `IsolatedStorage.Get(...)` look like a bare `Get(...)` on `Rec`, and the argument count (2 or 3) is compared against the primary-key field count.

Expected: no diagnostic. The fallback should only apply when the invoked method actually belongs to the record/table type (e.g. check that the invoked method symbol's containing type is the table / a record type, or that the invocation syntax has no qualifying expression). The same pattern probably affects other analyzers that adopted `GetReceiverTableType` in #509, and any other static built-in whose method shares a name with a record method (e.g. `IsolatedStorage.Delete`, `IsolatedStorage.Contains` vs. PC rules on `Delete`).

Inside a codeunit the same statement is **not** flagged (containing type is not a table), which confirms the fallback path.

### AL code to reproduce

```al
table 50100 "My Setup"
{
    fields
    {
        field(1; "Primary Key"; Integer) { }
    }
    keys
    {
        key(PK; "Primary Key") { Clustered = true; }
    }

    procedure GetPassword(): Text
    var
        Password: Text;
    begin
        // PC0013 fires here with 1.3.0: "Too many arguments provided; expected 1 arguments."
        if IsolatedStorage.Contains('MyKey', DataScope::Module) then
            IsolatedStorage.Get('MyKey', DataScope::Module, Password);
        exit(Password);
    end;
}
```

`IsolatedStorage.Get('MyKey', Password)` (2 arguments) is flagged as well.

### Additional context

- ALCops.Analyzers 1.2.0: no diagnostic. 1.3.0: error as above. Compiler: `microsoft.dynamics.businesscentral.development.tools` 17.0.34.45391 (BC 27.5, net8.0), run via ALOps v3 `ALOpsAppCompilerV3` with `ALCopsDownloadAnalyzers@1` (`version: latest`).
- Related: #509 (receiver-form audit that introduced `GetReceiverTableType`), #545 (follow-up audit; its table lists "Bare self → `Instance == null` inside tables", but static built-in types also produce `Instance == null` and are not covered).
- Workaround we use: pin `ALCopsDownloadAnalyzers@1` to `version: '1.2.0'`.

**Comment by Arthurvdv** [comment]
I’ve just released a bugfix for this in `v1.3.1`, where this issue should now be resolved.

This is indeed a regression that I somehow didn’t encounter during the beta release of `v1.3.0` _(and I’m hoping for more beta testers in the future to help me catch these)_

**Comment by JohnnyUndercover** [comment]
Thanks for the quick turnaround! I can confim that it is fixed in `v1.3.1`.

Removed our temporary `version: '1.2.0'` pin, the pipeline resolved `latest`
to `1.3.1` and the same app that failed this morning now compiles cleanly:

**Comment by Arthurvdv** [comment]
No problem, thanks for noticing and creating an issue! 🤗
